cachito/web/api_v1.py

```diff
@@ -33,14 +36,54 @@ def get_request(request_id):
     :return: a Flask JSON response
     :rtype: flask.Response
     :raise NotFound: if the request is not found
-    :raise ValidationError: if the request ID is of invalid format
     """
-    try:
-        request_id = int(request_id)
-    except ValueError:
-        raise ValidationError(f'{request_id} is not a valid request ID')
+    return flask.jsonify(Request.query.get_or_404(request_id).to_json())
+
+
+@api_v1.route('/requests/<int:request_id>/download', methods=['GET'])
+def download_archive(request_id):
+    """
+    Download archive of source code.
+
+    :param int request_id: the value of the request ID
+    :return: a Flask send_file response
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found
+    """
     request = Request.query.get_or_404(request_id)
-    return flask.jsonify(request.to_json())
+    # TODO: Verify request has already been processed.
+
+    cachito_shared_dir = flask.current_app.config.get('CACHITO_SHARED_DIR')
+    wait_timeout = flask.current_app.config.get('CACHITO_WAIT_TIMEOUT')
```

Comment:
Optional: Since the base configuration has a default value, you could directly access the value instead of using `.get` to show that it is expected that that value will always be defined. It's just a minor suggestion to improve readability.